alpha/action/migrate_test.go

```diff
@@ -0,0 +1,336 @@
+package action_test
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+	"github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+)
+
+func TestMigrate(t *testing.T) {
+	type spec struct {
+		name          string
+		migrate       action.Migrate
+		expectedFiles map[string]string
+		expectErr     error
+	}
+
+	sqliteBundles := map[image.Reference]string{
+		image.SimpleReference("test.registry/foo-operator/foo-bundle:v0.1.0"): "testdata/foo-bundle-v0.1.0",
+		image.SimpleReference("test.registry/foo-operator/foo-bundle:v0.2.0"): "testdata/foo-bundle-v0.2.0",
+		image.SimpleReference("test.registry/bar-operator/bar-bundle:v0.1.0"): "testdata/bar-bundle-v0.1.0",
+		image.SimpleReference("test.registry/bar-operator/bar-bundle:v0.2.0"): "testdata/bar-bundle-v0.2.0",
+	}
+
+	fooCatalog := `---
```

Comment:
I know we don't like local test files much but I actually prefer this to be a separate file than a long and hairy embedded text.

Reply:
I agree these interrupt the flow of the test as local variables. Rather than put them in separate testdata files where it's more difficult to tell where (and if) they're used, I made new functions at the bottom of the migrate_test.go file that just return them.